numba/core/dispatcher.py

```diff
@@ -853,6 +853,12 @@ def _rebuild(cls, uuid, py_func, locals, targetoptions,
         for sig in sigs:
             self.compile(sig)
         self._can_compile = can_compile
+        print("--py_func---", hex(id(py_func)))
+        print("--py_func.__globals__---", hex(id(py_func.__globals__)))
+        print("____", py_func.__globals__.keys())
+        # print(self.py_func.__globals__.keys())
+        self.__globals__ = py_func.__globals__
```

Comment:
Seems like this is the other change. It would be good to understand what code paths trigger this.